cengage/react-magma
# Chore: Improve TypeScript, Linting, and Styling Support in Doc MDX Files

Issue: #1553 

## What I did
   - Updated ESLint configuration `.eslintrc.js` to include `@typescript-eslint/parser` and `mdx` plugins.
   - Added rules to handle MDX files and TypeScript.
   - Disabled specific rules for MDX files to avoid unnecessary warnings.
   - Added `lint`, `lint:fix`, and `lint:check` scripts for consistent linting.
   - Updated `format` script to include more file type

## Screenshots:
![Screenshot from 2024-11-22 15-03-38](https://github.com/user-attachments/assets/599e93fc-1d57-4167-af49-c4112b6a33ac)

## Checklist 

## How to test
- Run linting and formatting scripts
- Ensure MDX files are properly linted and formatted.
- Check for any linting or formatting errors.
